src/modules/ui/tpApprovalFlow/reducer.ts

```diff
@@ -0,0 +1,26 @@
+import { APPROVAL_FLOW_UPDATE_PROGRESS, UpdateApporvalFlowProgressAction } from './action'
+
+export type TPApprovalFlowState = {
+  progress: number
+}
+
+export const INITIAL_STATE: TPApprovalFlowState = {
+  progress: 0
+}
+
+type TPApprovalFlowReducerAction = UpdateApporvalFlowProgressAction
+
+export const TPApprovalFlowReducer = (state = INITIAL_STATE, action: TPApprovalFlowReducerAction): TPApprovalFlowState => {
```

Comment:
What if we create a UI module for third parties in general and we put everything we need in there? We'll surely need  another progress bar for pushing changes.